src/FieldEncryptMapBase.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\pants\FieldEncryptMapBase.
+ */
+
+namespace Drupal\field_encrypt;
+
+use Drupal\Core\Plugin\ContextAwarePluginBase;
+
+/**
+ * This class structure mirrors the way blocks are constructed.
+ * We have a base class and a plugin interface.
+ *
+ * @package Drupal\field_encrypt
+ */
+abstract class FieldEncryptMapBase extends ContextAwarePluginBase implements FieldEncryptMapPluginInterface/*, ContainerInjectionInterface*/ {
+
+  protected $container;
+
+  public function __construct($configuration, $plugin_id, $plugin_definitions) {
+    parent::__construct($configuration, $plugin_id, $plugin_definitions);
+    $this->container = \Drupal::getContainer();
```

Comment:
What's your main driver for the container here? I don't see an immediate need for it.